### PR TITLE
CB-19260 pre stopstart scaling checks - validate health of dependent services in downscale flow

### PR DESCRIPTION
**Pull request by Vatsal23082000**
- Here we will be checking the health of the hosts on which the services will be dependent before the downscale operation

See detailed description in the commit message.

### CODE REVIEW

File: core/src/test/java/com/sequenceiq/cloudbreak/service/stack/flow/UpdateNodeCountValidatorTest.java [hunk 1]
```diff
@@ -189,9 +189,9 @@ public void testValidateStatusForStartHostGroup(
 
         if (errorMessageSegment.isPresent()) {
             checkExecutableThrowsException(errorMessageSegment, stackdto,
-                    () -> underTest.validateStackStatusForStartHostGroup(stackdto, instanceGroupAdjustmentV4Request));
+                    () -> underTest.validateStackStatusForStopStartHostGroup(stackdto, instanceGroupAdjustmentV4Request.getInstanceGroup()));
```
Comment (aakashshankar) [inline core/src/test/java/com/sequenceiq/cloudbreak/service/stack/flow/UpdateNodeCountValidatorTest.java:192]: Can a unit test be added to check if exceptions are thrown / not thrown when dependent services are unhealthy / healthy during downscale?

Comment (Vatsal23082000) [reply core/src/test/java/com/sequenceiq/cloudbreak/service/stack/flow/UpdateNodeCountValidatorTest.java:192]: Okay, I will add unit tests for downscale operation.

File: core/src/main/java/com/sequenceiq/cloudbreak/service/stack/flow/UpdateNodeCountValidator.java [hunk 2]
```diff
@@ -155,7 +155,7 @@ public void validateStackStatusForStartHostGroup(StackDto stack, InstanceGroupAd
             if (!unhealthyHostGroupNames.isEmpty()) {
                 throw new BadRequestException(format("Upscaling is Not Allowed for HostGroup: '%s' as Data hub '%s' has " +
```
Comment (sidseth) [inline core/src/main/java/com/sequenceiq/cloudbreak/service/stack/flow/UpdateNodeCountValidator.java:156]: This message needs to be downscale / upscale dependent.

Comment (Vatsal23082000) [reply core/src/main/java/com/sequenceiq/cloudbreak/service/stack/flow/UpdateNodeCountValidator.java:156]: Yes, I will do the change.

**Review by sidseth — APPROVED** [review]

**Review by lnardai — APPROVED** [review]

**Pull request merged**